src/lib.rs

```diff
@@ -136,6 +136,18 @@ pub enum Litep2pEvent {
     },
 }
 
+/// Configuration for adding protocols at runtime (protocols that are started after the Litep2p
+/// instance is created).
+struct RuntimeConfigs {
+    /// The keep-alive timeout for protocols.
+    ///
+    /// Needed for supporting protocols at runtime.
+    keep_alive_timeout: std::time::Duration,
+
+    /// Executor used to run protocols.
+    executor: Arc<dyn crate::executor::Executor>,
```

Comment:
nit: IMO `executor` deserves a dedicated field in `Litep2p`, and may be we don't need a new struct then.